raijin_scripts/deploy/build_environment_module.py

```diff
@@ -158,52 +277,99 @@ def install_pip_packages(pip_conf, variables):
         raise Exception('Either prefix: <prefix path> or target: <target path> is required by install_pip_packages:')
 
     LOG.debug(f'Installing pip packages from [ %s ] into directory [ %s ]', requirements, dest)
-    run(f'{pip} install -v --no-deps {arg} --compile --requirement {requirements}')
+    run_command(f'{pip} install -v --no-deps {arg} --compile --requirement {requirements}')
 
 
 def find_default_version(module_name):
+    """
+    Find default version of the dea module
+
+    :param module_name: Module name
+    :return: Version on success else raise exception
+    """
     cmd = f"module --terse avail {module_name}"
+    LOG.debug("Running command: %s", cmd)
     output = subprocess.run(cmd, shell=True, check=True,
                             stdout=subprocess.PIPE,
                             stderr=subprocess.STDOUT,
+                            universal_newlines=True,
                             encoding='ascii')
     versions = [version for version in output.stdout.splitlines() if f'{module_name}/' in version]
     default_version = [version for version in versions if '(default)' in version]
-    ret_val = None
+
     if default_version:
         ret_val = default_version[0].replace('(default)', '')
     elif len(versions) > 0:
         ret_val = versions[-1]
     else:
         raise Exception('No version of module %s is available.' % module_name)
-
     return ret_val
 
 
-def run_final_commands_on_module(commands, module_name):
+def run_final_commands_on_module(commands, module_path):
+    """
+    Run final commands as per configurations in modulespec.yaml file
+
+    :param commands: List of commands
+    :param module_path: Module path
+    :return: None
+    """
     for command in commands:
-        command = f'module load {module_name}; {command}'
-        run(command)
+        cmd = f'{module_path}/bin/{command}'
+        run_command(cmd)
 
 
 def include_stable_module_dep_versions(config):
+    """
+    Include stable module dependency versions
+
+    :param config: Dictionary of configuration variables
+    :return: None
+    """
     stable_module_deps = config.get('stable_module_deps', [])
     for dep in stable_module_deps:
         default_version = find_default_version(dep)
         dep = dep.replace('-', '_')
         config['variables'][f'fixed_{dep}'] = default_version
 
 
-def main(config_path):
+def reinstall_miniconda(script_name):
+    """
+    Re-install miniconda and npm packages
+
+    :param script_name: Shell script name
+    :return: None
+    """
+    LOG.debug('Re-install miniconda3 before creating new dea-environment module')
+    run_command(f'./{script_name}')
+
+
+def main(config_path, dea_env):
+    """
+    Build new environment module
+
+    :param config_path: Configuration path
+    :param dea_env: Are we building dea-env module?
```

Comment:
We shouldn't have this `dea_env` argument. It splits up the configuration and hard codes details that should be configurable. It should be included in the configuration file OR we should just reinstall miniconda if we're going to be using `conda`.

Reply:
`reinstall_miniconda_package: reinstall_miniconda.sh` was added to configuration file and remove `dea_env` argument. Updated README.md accordingly.